### PR TITLE
Foreign Library RTS Selection

### PR DESCRIPTION
**Pull request by TravisWhitaker**
Fresh PR for https://github.com/haskell/cabal/pull/5118

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

### CODE REVIEW

**Comment by sergv** [comment]
@23Skidoo Could we merge this, please? The prerequisite #5118 was closed stating that something's merged, presumably this PR, but it looks like it isn't.

**Comment by 23Skidoo** [comment]
@sergv Thanks for the reminder, for some reason I believed that this was already merged.

**Comment by 23Skidoo** [comment]
Also cherry-picked into `2.4`.

File: Cabal/Distribution/Simple/Configure.hs [hunk 1]
```diff
@@ -1979,9 +1976,6 @@ checkForeignLibSupported comp platform flib = go (compilerFlavor comp)
 
     goGhcLinux :: ForeignLibType -> Maybe String
     goGhcLinux ForeignLibNativeShared
-      | standalone = unsupported [
```
[the remark this answers is not in the document]
Comment (TravisWhitaker) [reply Cabal/Distribution/Simple/Configure.hs]: This build type is supported in general, just not by the GHC you have installed.

**Pull request merged**